### PR TITLE
Quiche roll 20220407195958

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Update QUICHE from f9c571cac to bac04054b
https://github.com/google/quiche/compare/f9c571cac..bac04054b
    
$ git log f9c571cac..bac04054b --date=short --no-merges --format="%ad %al %s"
    
2022-04-07 bnc Relocate QUICHE files into quiche/ directory within the quiche repo, and change the relative include paths accordingly.
2022-04-07 ianswett Set the chrome_value to true for quic_allow_client_enabled_bbr_v2 to allow the flag to be deprecated.
2022-04-07 haoyuewang Rollback --gfe2_reloadable_flag_quic_set_burst_token as the GWS experiment completes.
2022-04-07 quiche-dev Disable test violating stringview [preconditions](https://timsong-cpp.github.io/cppwp/string.view.cons).
2022-04-07 haoyuewang Add SendAlgorithmInterface::HasGoodBandwidthEstimateForResumption() to be used to gauge if there is a good enough bandwidth estimate for bandwidth resumption.
2022-04-06 fayang Simply SetRetransmissionAlarm by consolidating more of the changes from cl/381566145 and cl/314387174 into SetRetransmissionAlarm to ensure the logic is applied consistently.
    
Signed-off-by: Ryan Hamilton <rch@google.com>

### CODE REVIEW

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -850,9 +850,16 @@ def _com_googlesource_chromium_zlib():
     )
 
 def _com_github_google_quiche():
-    external_genrule_repository(
+    external_http_archive(
         name = "com_github_google_quiche",
-        genrule_cmd_file = "@envoy//bazel/external:quiche.genrule_cmd",
+        # Rewrite third_party includes and #pragma clang.
```
Comment (DavidSchinazi) [inline bazel/repositories.bzl:855]: Out of curiosity, why is this here instead of in the genrule file?

**Review by DavidSchinazi — APPROVED** [review]
Woohoo, code deletion!

File: bazel/repositories.bzl (hunk 1 above)
Comment (RyanTheOptimist) [reply bazel/repositories.bzl:855]: Heh, you must have miss my whining on Chat :) The genrule approach works by taking a list of input files and transforming them into a distinct of output files. The tip-of-tree genrule moves the input files into a quiche/ directory. But since we actually moved the files in the underlying quiche repo, envoy no longer needs to do this.As a result, genrule no longer works. :( The quiche.genrule_cmd file is hence deleted in this PR. Thankfully, this "patch_cmds" approach does seem to work.

Comment (DavidSchinazi) [reply bazel/repositories.bzl:855]: Ah right, thanks!

**Comment by RyanTheOptimist** [comment]
@mattklein123 mind taking taking a quick look?

**Review by mattklein123 — APPROVED** [review]
